projecttacoma/deqm-test-server-frontend
# Calculate evaluate measure

## Summary
Implementation of Evaluate Measure calculation (request and response handling). 
## New behavior
- A Calculate button is rendered on any Evaluate Measure page. It is only enabled if the request is valid. 
- If the Calculate button is pressed, a request is sent to the test server using the url that has been "built" by the user. 
- If the measure evaluation is successful, a success notification appears and the MeasureReport is displayed in a Prism component to the right of the Calculate button. 
- If the measure evaluation is unsuccessful, an error notification appears.

## Code changes
- **pages/[resource]/[id]/evaluate.tsx** now renders the Calculate button, Prism component, and notifications
- **__tests__/pages/resource/id/evaluate.test.tsx** unit tests added

## Testing Guidance
- See the README.md for first time setup instructions.
- Be sure to have the latest version of DEQM Test Server running
#### Test in a browser
- To start the frontend: `npm run dev` then navigate to http://localhost:3001 in a browser
1. Successful measure evaluation:
    - Choose a valid set of options to build the evaluate measure request:
      a. Choose reportType: Subject, then select a Patient  
               OR
      b. Chose reportType: Population
    - Click the Calculate button. A success notification and a Prism component with the MeasureReport should both render
2. Failed measure evaluation:
    - Choose reportType: Population, then select a Practitioner which has no Patient resources.
    - Click the Calculate button and an error notification should appear, specifying the issue.

#### Unit tests
- Run the unit tests: `npm run test`
